src/foam/dao/MDAO.java

```diff
@@ -165,7 +161,14 @@ public Sink select_(X x, Sink sink, long skip, long limit, Comparator order, Pre
 
   public void removeAll_(X x, long skip, long limit, Comparator order, Predicate predicate) {
     synchronized ( lock_.writeLock() ) {
-      state_ = null;
+
```

Comment:
This should be:
if ( predicate == null ) { state_ = null } else { super.select_(x, skip, limit, order, predicate); }

Reply:
@kgrgreer thank you very much for your help.